### PR TITLE
Deny field access to imported fields on nodes

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Took the similar funcitonality from stanford_person_importer and generalized it.
   - https://github.com/SU-SWS/stanford_person/blob/8.x-1.x/modules/stanford_person_importer/stanford_person_importer.module#L121

# Need Review By (Date)
- :shrug: 

# Urgency
- low

# Steps to Test
1. checkout this branch and clear caches
2. import some content like people or events
3. edit the person or event and verify you don't see fields that are imported.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by imonroe** [comment]
Ok, I have a question on this vis-a-vis the event importer.  I've got a task to make it so that the localist URL is mapped to the external source field in the event which is created.  That turns the event listing into an external link, and visitors to the site are sent directly to localist, instead of the event page on the drupal side.  

If the users wanted to, they'd be able to remove that imported localist link in order to open up the page on the drupal side, and add additional information if they want.  However, if that field is forbidden, they won't be able to make that change, and all localist imported events will go to Localist pages regardless.  Is that the behavior we want?  cc @ishahism ....

**Comment by pookmish** [comment]
If they were allowed to edit the field, their change would be lost on the next import anyways so there's not much that can be done there with or without this PR.

**Comment by pookmish** [comment]
Correct, but a user will loose their changes at some point whether its in 6 hours or 20 days. There's very likelihood of the event being modified on Localist and we should never step on a user's toes.

File: stanford_migrate.module [hunk 1]
```diff
@@ -33,6 +38,93 @@ function stanford_migrate_help($route_name, RouteMatchInterface $route_match) {
...
+ *   Migration entity or null/false if none found.
+ */
+function stanford_migrate_get_migration(NodeInterface $node) {
+  // Use a static variable so that it doesn't look up the migrations multiple
+  // times.
+  $node_migration = &drupal_static(__FUNCTION__ . $node->id());
+  if (!is_null($node_migration)) {
+    return $node_migration;
+  }
+  // Se the static to false and check for null above. If the first attempt to
```
Comment (imonroe) [inline stanford_migrate.module:57]:
```suggestion
  // Set the static to false and check for null above. If the first attempt to
```

File: stanford_migrate.module [hunk 2]
```diff
@@ -33,6 +38,93 @@ function stanford_migrate_help($route_name, RouteMatchInterface $route_match) {
...
+
+    // This will check if a migrate process is mapped to a specific column on
+    // the field.
+    foreach (array_keys($columns) as $column) {
+      $processing = $processing ?: !empty($migration->process["$field_name/$column"]);
+    }
+
+    if ($processing) {
+      \Drupal::messenger()
+        ->addWarning(t('Some fields can not be edited since they contain imported & synced data. They are not visible here.'));
```
Comment (imonroe) [inline stanford_migrate.module:119]: this is smart.

**Review by imonroe — APPROVED** [review]
This works as advertised.  I'm glad you've included a message to the user when editing the node:

image

One tiny typo, but other than that, good to go.

**Comment by imonroe** [comment]
Also, great inline comments, it made it very clear what was going on.

**Pull request merged**